wcfsetup/install/files/lib/system/flood/FloodControl.class.php

```diff
@@ -0,0 +1,241 @@
+<?php
+namespace wcf\system\flood;
+use wcf\data\object\type\ObjectTypeCache;
+use wcf\system\SingletonFactory;
+use wcf\system\WCF;
+use wcf\util\DateUtil;
+use wcf\util\UserUtil;
+
+/**
+ * Flood control tracks the times contents are created by users.
+ * 
+ * @author      Matthias Schmidt
+ * @copyright   2001-2020 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package     WoltLabSuite\Core\System\Flood
+ * @since       5.4
+ */
+class FloodControl extends SingletonFactory {
+	const PRUNE_TIME = TIME_NOW - 31 * 86400;
+	
+	/**
+	 * Returns the number of contents by a certain identifier type within a certain
+	 * period of time `[$time-$interval, $time]` and the earliest time within the period content
+	 * was created.
+	 */
+	protected function countContentByIdentifier(string $objectType, string $identifier, \DateInterval $interval, int $time): array {
+		$sql = "SELECT  COUNT(*) AS count, MIN(time) AS earliestTime
+			FROM    wcf" . WCF_N . "_flood_control
+			WHERE   objectTypeID = ?
+			        AND identifier = ?
+			        AND time >= ?
+			        AND time <= ?";
```

Comment:
Should this be a closed interval? I would expect an half-open interval:

I'd also like to suggest a slight reformatting to make the interval more clear:
```suggestion
			        AND (? <= time AND time < ?)";
```